test/integ/pandas/test_pandas_tools.py

```diff
@@ -416,6 +416,297 @@ def mocked_execute(*args, **kwargs):
             )
 
 
+@pytest.mark.parametrize("quote_identifiers", [True, False])
+@pytest.mark.parametrize("auto_create_table", [True, False])
+def test_stage_location_building_db_schema(conn_cnx, quote_identifiers: bool, auto_create_table: bool):
+    """This tests that write_pandas constructs stage location correctly with database and schema."""
+    from snowflake.connector.cursor import SnowflakeCursor
+
+    with conn_cnx() as cnx:
+
+        def mocked_execute(*args, **kwargs):
+            if len(args) >= 1 and args[0].lower().startswith(('create temporary stage', 'put', 'copy into', 'select')):
+                stage_location = None
+                if args[0].lower().startswith('create temporary stage'):
```

Comment:
Thank you for the detailed test 👍 It's also acceptable to only test the case for `COPY INTO`, which is what the `table_location` test does.

Reply:
I think it might actually be better if we only test one query, both for simplicity and in the case if we need to tweak some of the queries, we can minimize the change to these tests.

Reply:
It was refactored to test only creation query for all 3 cases (table, stage, file format) [here](https://github.com/snowflakedb/snowflake-connector-python/pull/1274/commits/331b999014708427e25e8d11a3ce0aea5642cb58).